### ISSUE TITLE
Not working on Ubuntu 17.10 with CustomCorner extension.

### ISSUE THREAD

**Issue by 16NoCo** [LinxGem33/Arc-Menu#194]
I recently discovered this extension, so I don't know if it works with something else than Ubuntu 17.10, but when I click on the Applications (Arc) icon nothing happens.

If I disable the CustomCorner extension it works fine.

**Comment by LinxGem33** [comment]
@16NoCo 

Can you please provide some logs that capture the occurrence of the bug? For example:
```
journalctl /usr/bin/gnome-shell -f > arc-menu-log.txt
```
For more details see:
https://github.com/LinxGem33/Arc-Menu/wiki/Supporting-Information

**Comment by 16NoCo** [comment]
This is the log after restarting the Gnome-Shell, with only Arc-Menu and CustomCorner enabled.
Triggering a hot corner and clicking the Arc-Menu button added nothing to the log.

``` Window manager warning: Overwriting existing binding of keysym ff09 with keysym ff09 (keycode 17).
JS WARNING: [/home/noe/.local/share/gnome-shell/extensions/customcorner@eccheng.gitlab.com/extension.js 110]: reference to undefined property Symbol.toPrimitive
Window manager warning: Overwriting existing binding of keysym ff09 with keysym ff09 (keycode 17).
JS WARNING: [resource:///org/gnome/shell/ui/modalDialog.js 218]: reference to undefined property "GdkX11Screen"
Window manager warning: CurrentTime used to choose focus window; focus window may not be correct.
Window manager warning: Overwriting existing binding of keysym ff09 with keysym ff09 (keycode 17).
JS WARNING: [resource:///org/gnome/shell/ui/main.js 315]: reference to undefined property "MetaStage"
JS WARNING: [resource:///org/gnome/shell/ui/layout.js 217]: reference to undefined property "MetaWindowGroup"
JS WARNING: [resource:///org/gnome/shell/ui/osdMonitorLabeler.js 59]: reference to undefined property "MetaDBusDisplayConfigSkeleton"
JS WARNING: [resource:///org/gnome/gjs/modules/tweener/tweener.js 540]: reference to undefined property "isSpecialProperty"
current session already has an ibus-daemon.
JS WARNING: [resource:///org/gnome/shell/ui/slider.js 38]: reference to undefined property "CallyActor"
Telepathy is not available, chat integration will be disabled.
Some code accessed the property 'DBusMenu' on the module 'interfaces'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Some code accessed the property 'WORKAROUND_RELOAD_TYPE_REGISTER' on the module 'util'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Some code accessed the property 'StatusNotifierWatcher' on the module 'statusNotifierWatcher'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Some code accessed the property 'StatusNotifierWatcher' on the module 'interfaces'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
JS WARNING: [/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/appIcons.js 1361]: unreachable code after return statement
JS WARNING: [/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/utils.js 34]: reference to undefined property Symbol.toPrimitive
WARNING: addSignalMethods is replacing existing [0x5588a9e51280 Gjs_ShowAppsIcon.dash-item-container] connect method
WARNING: addSignalMethods is replacing existing [0x5588a9e51280 Gjs_ShowAppsIcon.dash-item-container] disconnect method
WARNING: addSignalMethods is replacing existing [0x5588a9e51280 Gjs_ShowAppsIcon.dash-item-container] emit method
Window manager warning: Overwriting existing binding of keysym ff09 with keysym ff09 (keycode 17).
Some code accessed the property 'Logger' on the module 'util'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
[AppIndicatorSupport-WARN] Failed to acquire org.kde.StatusNotifierWatcher
Window manager warning: "XF86RFKill" is not a valid accelerator
Window manager warning: Overwriting existing binding of keysym ff09 with keysym ff09 (keycode 17).
Error looking up permission: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.impl.portal.PermissionStore was not provided by any .service files
Some code accessed the property 'State' on the module 'docking'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Window manager warning: Overwriting existing binding of keysym ff09 with keysym ff09 (keycode 17).
GNOME Shell started at Mon Oct 23 2017 11:06:38 GMT+0200 (CEST)```

**Comment by lexruee** [comment]
@16NoCo 
Thanks for the bug report and the logs. I guess Arc-Menu and CustomCorner are interfering in some way or another. To find out the problem we need to have a look at the source code of the CustomCorner extension.

**Comment by 16NoCo** [comment]
For information, I get the same issue with the Applications Menu extension, and this extension was interfering with No Topleft Hot Corner. Since then No Topleft Hot Corner got an update which resolves the issue.
Here is the patch that could be applied to Applications Menu before the update, if it can help in any way : https://bugzilla.gnome.org/show_bug.cgi?id=712625

**Comment by lexruee** [comment]
@16NoCo Thanks for the link to the bug report. It made my day :-)

@LinxGem33 FYI: This might be offtopic. Anyways, I really like the comment of the GNOME Shell dev Florian Müllner:

```
(In reply to Matt Mower from comment #3)
> I'm resurrecting this because the issue just hit me on Ubuntu 17.10 with
> gnome-shell 3.26.

With *Ubuntu-patched* gnome-shell to be exact. To be honest, if a distribution changes 
gnome-shell in a way that breaks extensions, then that's first and foremost a problem of 
the distribution. extensions.gnome.org is supposed to be distro-agnostic, so given that 
we don't have resources to test on all distros (or even all major ones), the only sane 
approach is to follow upstream. So frankly, while bug 688320 is still under discussion 
I'm not very inclined to make any changes to the extension ...
```
It really expresses my attitude towards Ubuntu 17.10...

**Comment by lexruee** [comment]
@16NoCo 
I created the branch `ubuntu-fixes` on my repo which adds back the hotcorner code from upstream as well as the mentioned sumitted patch in the bug report: https://bugzilla.gnome.org/show_bug.cgi?id=712625

Here is my repo: https://github.com/lexruee/Arc-Menu/tree/ubuntu-fixes

Let me know it fixes the problem. Thanks in advance :-).

**Comment by 16NoCo** [comment]
- I noticed many people having this kind of issue (the one Florian Müllner replied to) with Ubuntu 17.10 GNOME Shell, but I don't get this issues (maybe because I already installed GNOME in the past). Anyway I think I'll use the Vanilla Shell since I have no use of Ubuntu Dock (or the rest they modified in the GNOME).

- Sadly your fix doesn't work for me. I don't know if it's related but here's what I did : 

1. Disabled the original extension (from extensions.gnome.org) ;
2. Downloaded and extracted your fix from ZIP file (Should I have used the `git clone` command instead ?) ;
3. `make uninstall` ;
4. `make install` ;
5. `make enable` ;
6. Reloaded the GNOME Shell (<kbd>alt</kbd>+<kbd>f2</kbd> then `r`).

For 3. to 5. I get 3 times this line
`fatal: Not a git repository (or any of the parent directories): .git`
but the installation seems to complete : I still can't click the button but <kbd>alt</kbd>+<kbd>f1</kbd> and <kbd>Super</kbd>+<kbd>x</kbd> both open the Arc-Menu (just as before).

**Comment by lexruee** [comment]
Hmm, are you using the Ubuntu session? I don't think that Arc-Menu works with the ubuntu-session (i.e., default session). So far I only tested it with the gnome-session.

In case you use the gnome-session on Ubuntu, can you test it again by using git clone?

**TL;DR:**
A common git workflow to test a branch is as follows:

1) Git clone the git repo and checkout a specific branch like `ubuntu-fixes`:

```
git clone https://github.com/lexruee/Arc-Menu --branch ubuntu-fixes
```

2) Change into the Arc-Menu directory, run `make uninstall` to remove the existing installation, and run `make install` to install locally Arc-Menu (i.e.,  `~/.local/share/gnome-shell/extensions`).

In other words:
```
cd Arc-Menu
make uninstall
make install
```

3) Restart GNOME Shell.

**Comment by 16NoCo** [comment]
No, I'm not using the Ubuntu session since I checked earlier for the log (I select on 'GNOME on Xorg' on the log-in screen). I tried with the `git clone` command as you said, but the result is the same, except I don't get the 3 fatal error lines any more.

**Comment by LinxGem33** [comment]
@16NoCo 

 appreciate the feed and logs, after further investigation into this issue i found out that there are few issues with custom corner extension  the biggest being that it does not currently officially support gnome shell 3.26 and just changing the metadata.js file does not necessarily offer proper support as the gnome team currently employed a strict mode in the gnome extension code development which involved most extension developers to rework their code base to confirm to this new standard also looking at the customer corner code it hasn’t been updated in over a year... so i would advise you to ping the developer and inform him of your issue regarding the conflicts the extension is having with other extensions.

@lexruee

**Comment by 16NoCo** [comment]
Ok I'll do that. I first came here because the patch from the link I shared earlier was to implement in Applications Menu not in No Topleft Hot Corner, but it's true that since the issue comes from the other extension they should fix it.

Thanks for the help and the time.

P.S. : I was about to post that I found out that touching the Arc button on a touch-screen opens the menu, so the issue seems only to appear with a mouse...
